### DISCUSSION TITLE
Hide menubar icon when no external display is connected

### DISCUSSION THREAD

**Discussion by Muirium** [MonitorControl/MonitorControl#590] [Ideas]
Awesome app, loving it on my M1 Air! Thanks for making this!

A couple of ideas:

Could you allow an automatically toggle of the "Hide the menu icon" setting when running on a laptop with no external displays connected? I love seeing MonitorControl's icon up there on my 4K Dell display, which I run with my MacBook Air closed in clamshell. Really nice minimalist UI for finesse Brightness and Contrast controls by mouse. But that same menu is useless on my Air's own display (which I use when mobile), and there's less screen real-estate to spare for it on the 13" too! Smart auto-hide would give me the best of both behaviours.

Secondly: keyboard shortcuts (yay!) for Contrast as well as Brightness? I think I saw this is coming in 3.1, if so: great! My Dell needs both tweaked quite often, so keys are super for that. I'm thinking of Shift+F1/2 for Contrast in addition to straight F1/2 for Brightness control. I use vintage mechanical keyboards when I'm at my desk, so simple old school function row keys are perfect. And customisable keys are indeed sublime!

**Comment by fbirlik** [comment]
I would also appreciate this feature and it was actually mentioned before.  Let me see if I can link it.

https://github.com/MonitorControl/MonitorControl/issues/536#issuecomment-905385784

**Comment by Muirium** [comment]
+1ed

From the tags on that post, it looks like this feature is coming to version 3.1 too. Hope so! Read my mind!

**Comment by Muirium** [comment]
Useful menu:
![Screenshot 2021-09-14 at 8 56 24 pm](https://user-images.githubusercontent.com/221646/133448363-c59fa9c4-a46f-4910-b45c-8104fa613ef3.jpg)

Not so useful:
![Screenshot 2021-09-15 at 2 33 27 pm](https://user-images.githubusercontent.com/221646/133448426-8e471bb2-69f5-4965-a686-78ddd58363a7.jpg)

Auto-hide that last one and we're all set!

**Comment by Muirium** [comment]
Looks like internal display control is [coming in version 3.1](https://github.com/MonitorControl/MonitorControl/discussions/596). Cool!

Mind, Apple's built in display controls are fine for driving the internal display. I'm only really using MonitorControl when I've my desktop display hooked up. So I'd likely still want this hiding, to save me space on the not-so-roomy little 13 inch laptop.

**Comment by waydabber** [comment]
I reopened the request now since it looks like there is more demand for this option.

https://github.com/MonitorControl/MonitorControl/issues/536

**Comment by waydabber** [comment]
I added this one. See the updated screenshot here: https://github.com/MonitorControl/MonitorControl/discussions/596

**Reply by Muirium** [reply]
Niice! Thanks.
